js/angular-edrna/Reorder.js

```diff
@@ -0,0 +1,174 @@
+          var _handler = angular.element(e.event.target);
+          var _currentElement = angular.element(e.event.target);
+          var _allElements = $element.find('[reorder-element=\'' + scopeTag + '\']');
+          overallDiff = 0;
+
+          /* Set current element */
+          if(typeof _currentElement.attr('reorder-element') === 'undefined') {
+            var _parents = _currentElement.parents('[reorder-element]');
+
+            if(_parents.length > 0) _currentElement = angular.element(_parents[0]);
+            else {
+              $document.off('touchmove mousemove');
+              return false;
+            }
+          }
+          draggingElement = _currentElement;
+
+          if(useHandler) {
+            /* Validate handler click, with propagation possible */
+            if(typeof _handler.attr('reorder-handler') === 'undefined') {
+              var _parents = _handler.parents('[reorder-handler]');
+
+              if(_parents.length > 0) _handler = angular.element(_parents[0]);
+              else _handler = undefined;
+            }
+
+            /* Stop drag if using handler and handler not clicked with LMB */
+            if(typeof _handler === 'undefined' || e.event.button === 2) {
+              $document.off('touchmove mousemove');
+              return false;
+            }
+          }
+
+          /* Set current position */
+          currentPosition = _allElements.index(_currentElement);
+          originalElementPosition = currentPosition;
+          
+          /* When hover over another element */
+          _allElements.on('mouseenter.reorder', function(e) {
+            var _hoveredElement = angular.element(e.currentTarget);
+            var _allElements = $element.find('[reorder-element=\'' + scopeTag + '\']');
+            var hoveredPosition = _allElements.index(_hoveredElement);
+
+            var diff = hoveredPosition - currentPosition;
+            
+            if(diff < 0) moveUp(Math.abs(diff), _allElements);
+            if(diff > 0) moveDown(diff, _allElements);
+            currentPosition = hoveredPosition;
+          });
+        });
+
+        $scope.$on('draggable:move', function(scope, e) {
+          if(!dragInProgress) {
+            dummyElement = createDummy(draggingElement[0].tagName);
+            dummyElement.insertBefore(draggingElement);
+
+            dragInProgress = true;
+          }
+        });
+
+        $scope.$on('draggable:end', function(e, d) {
+          
+          $element.find('[reorder-element=\'' + scopeTag + '\']').off('mouseenter.reorder');
+          if(dummyElement) dummyElement.remove();
+
+          draggingElement = undefined;
+          dummyElement = undefined;
+          dragInProgress = false;
+          
+          if(!objectChanged) return;
+          else objectChanged = false;
+
+          $scope.reorderComplete(angular.copy($scope.reorderData), originalElementPosition, overallDiff);
+        });
+
+        var getElements = function(callback) {
```

Comment:
Why not `function getElements () {}`? That's much better than `var getElements = function () {}`. 

1. Hoisting
2. Easier to read due to different code highlight
3. Easier to debug due to function name in stack traces